tests/Tests/Functional/ShardKeyTest.php

```diff
@@ -35,7 +35,9 @@ public function setUp(): void
 
     public function tearDown(): void
     {
-        $this->logger->unregister();
+        if (isset($this->logger)) {
+            $this->logger->unregister();
+        }
```

Comment:
```suggestion
        $this->logger?->unregister();
```

Reply:
Your suggested code is not equivalent. `?->` works with `null` but not uninitialized properties. Not the same behavior as `??`.
https://3v4l.org/POp20X

Reply:
This would work, but it would be so much less readable.
```suggestion
        ($this->logger ?? null)?->unregister();
```

Reply:
Oh, yeah, kinda forgot those are a thing. Let's stick with your original solution then :)